tofhir-server/src/main/scala/io/tofhir/server/repository/schema/SchemaFolderRepository.scala

```diff
@@ -451,5 +469,49 @@ class SchemaFolderRepository(schemaRepositoryFolderPath: String, projectFolderRe
       throw BadRequest("Schema definition is not valid.", s"Schema definition type must starts with uppercase!");
     }
   }
+
+  /**
+   * Validates that all referenced profiles in a given schema either exist as base FHIR definitions,
+   * are already present in the system, or are included in the list of schemas to be created.
+   *
+   * @param projectId The ID of the project in which the schemas are being validated.
+   * @param schemaResource The FHIR StructureDefinition resource representing the schema to be validated.
+   * @param schemaUrls The list of URLs of the schemas that are included in the current batch to be created.
+   * @throws BadRequest if any referenced profile is missing.
+   */
+  private def validateReferencedSchemas(projectId: String, schemaResource: Resource, schemaUrls: Seq[String]): Unit = {
+    /**
+     * Validates a single profile URL to ensure it either exists as a base FHIR definition,
+     * is already present in the system, or is included in the provided schema URLs list.
+     *
+     * @param profile   The URL of the profile to be validated.
+     * @param schemaUrl The URL of the schema that references this profile, used for error reporting.
+     * @throws BadRequest if the profile is missing.
+     */
+    def validateProfile(profile: String, schemaUrl: String): Unit = {
+      if (!profile.startsWith("http://hl7.org/fhir/StructureDefinition") &&
```

Comment:
Maybe we can use onfhir API for these kind of constant strings such as StructureDefinition from io.onfhir.api.FHIR_FOUNDATION_RESOURCES.